src/kemal/route_handler.cr

```diff
@@ -37,18 +39,20 @@ module Kemal
       if route.found?
         @cached_routes.clear if @cached_routes.size == CACHED_ROUTES_LIMIT
         @cached_routes[lookup_path] = route
+        context.params = Kemal::ParamParser.new(request, route.params)
```

Comment:
Because the context no longer has a connection to the RouteHandler, the params are now set on the context. This is in a weird spot (`lookup_route`) because `FilterHandler` might need the params and uses this method.